sympy/matrices/matrices.py

```diff
@@ -2616,8 +2649,28 @@ def dot(self, b):
             mat = mat.reshape(1, n)
         if b.shape != (n, 1):
             b = b.reshape(n, 1)
+
         # Now ``mat`` is a row vector and ``b`` is a column vector.
-        return (mat * b)[0]
+
+        # If it so happens that only conjugate_convention is passed
+        # then automatically set hermitian to True. If only hermitian
+        # is true but no conjugate_convention is not passed then
+        # automatically set it to "maths"
+
+        if (conjugate_convention is not None) and (hermitian is None):
+            hermitian = True
+        if (hermitian == True) and (conjugate_convention is None):
+            conjugate_convention = "maths"
+
+        if hermitian == True:
+            if conjugate_convention in ("maths", "left", "math"):
+                return (mat.conjugate()*b)[0]
+            elif conjugate_convention in ("physics", "right"):
+                return (mat*b.conjugate())[0]
```

Comment:
I think this code would be better if you did `b=b.conjugate()` or `mat=mat.conjugate()` and then returned a value only once.